R/ts_gtrends_windows.R

```diff
@@ -23,13 +35,29 @@ ts_gtrends_windows <- function(keyword = NA,
                                retry = 5) {
   library(tidyverse)
 
-  x <- tibble(
-    start_date = seq(as.Date(from), by = stepsize, length.out = n_windows)
+    tbl <- prepare_windows_tbl(from, n_windows, prevent_window_shrinkage, stepsize, windowsize)
+    tbl <- tbl %>% mutate(trend_data = list(ts_gtrends(keyword, category, geo, window, retry, wait, quiet))) %>%
+    unnest(cols = trend_data) %>%
+    ungroup()
+
+    return(tbl)
+  # TODO: The problem to solve:
+  #       Every time 100 appears within stepsize of the start/end of a chunk
+  #       a wild rescaling
+}
+
+
+prepare_windows_tbl <- function(from, n_windows, prevent_window_shrinkage, stepsize, windowsize) {
+  # prepare tibble with single column *windows* which for each time-window contains a string with start-date
+  # and end-date
+  tbl <- tibble(
+  start_date = seq(as.Date(from), by = stepsize, length.out = n_windows)
   ) %>%
-    rowwise() %>%
-    mutate(end_date = seq(start_date, length.out = 2, by = windowsize)[2]) %>%
-    ungroup() %>%
-    # truncate end date to Sys.Date() to avoid google errors
+  rowwise() %>%
+  mutate(end_date = seq(start_date, length.out = 2, by = windowsize)[2]) %>%
+  ungroup()
+
+  tbl <- tbl[1:nrow(tbl),] %>%
```

Comment:
```r
mutate(end_date = as.Date(end_date)) %>%
```

is a bit cleaner and you don't need the intermediate `tbl`.

Reply:
unfortunately, this does not resolve the error, see here: https://github.com/trendecon/trendecon/issues/3#issuecomment-632089964

Reply:
Sorry, this one:

    mutate(date = as.Date(as.numeric(date), origin = "1970-01-01")) %>%

Reply:
Indeed. Fixed here d7b9734